packages/kit/src/runtime/server/utils.js

```diff
@@ -97,17 +97,20 @@ export async function handle_fatal_error(event, options, error) {
 export async function handle_error_and_jsonify(event, options, error) {
 	if (error instanceof HttpError) {
 		return error.body;
-	} else {
-		if (__SVELTEKIT_DEV__ && typeof error == 'object') {
-			fix_stack_trace(error);
-		}
+	}
 
-		return (
-			(await options.hooks.handleError({ error, event })) ?? {
-				message: event.route.id != null ? 'Internal Error' : 'Not Found'
-			}
-		);
+	if (__SVELTEKIT_DEV__ && typeof error == 'object') {
+		fix_stack_trace(error);
 	}
+
+	const is_runtime_error =
+		error instanceof TypeError || error instanceof SyntaxError || error instanceof ReferenceError;
```

Comment:
I'm not sure if this is an exhaustive list or if it'd be possible to create one. Perhaps we could go in the other direction? What the type of the error when a 404 is returned without additional error? Could we check if it's that type?

Reply:
iirc we don’t have a type for that right now so we'd have to create one

Reply:
It looks like we get to this 500 error from here: https://github.com/sveltejs/kit/blob/dbbd4c7fc1ae2abef584a29a688e2247a80b9792/packages/kit/src/runtime/server/page/respond_with_error.js#L108

The 404 error happens here: https://github.com/sveltejs/kit/blob/dbbd4c7fc1ae2abef584a29a688e2247a80b9792/packages/kit/src/runtime/server/page/respond_with_error.js#L92C63-L92C63

I wondered if perhaps the error message should get handled in that line 108 in response_with_error.